client/state/selectors/get-media.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Returns media for a specified site ID and query.
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {Object}  query  Query object
+ * @return {?Array}         Media
+ */
+export default function getMedia( state, siteId, query ) {
+	const queries = state.media.queries[ siteId ];
+
+	if ( ! queries ) {
+		return null;
+	}
+
+	return queries.getItems( query );
```

Comment:
Also, in case it lends anything to the argument, operations on the QueryManager class are immutable, i.e. one can rely upon strict equality between two instances to determine whether the contents are in-fact equal.

https://github.com/Automattic/wp-calypso/blob/31ff257/client/lib/query-manager/test/index.js#L199-L283